src/modules/trajectory_generation/interface/src/rviz_visualization.cpp

```diff
@@ -41,8 +35,21 @@ RvizPublisher::RvizPublisher(rclcpp::Node* node,
 
 }
 
-void RvizPublisher::publish()
-{
+void RvizPublisher::publish(const common_ros2::msg::Trajectory& trajectory)
+{ 
+  geometry_msgs::msg::Point start_point;
+  start_point.x = trajectory.trajectory_points[0].x;
+  start_point.y = trajectory.trajectory_points[0].y;
+  start_point.z = trajectory.trajectory_points[0].v;
+  points.points.push_back(start_point);
+
+  for (unsigned int i = 0; i < trajectory.trajectory_points.size(); i++) {
```

Comment:
Since we do not need indices in this iteration I think `for(const auto &pt:trajectory_points)` would be faster.

Thoughts? @gprajwalpoojari

Reply:
Yes, that is possible. Please update.

Reply:
Agreed

Reply:
Done.

Reply:
Also modified clock to node->get_clock()